src/object/srv_ec_aggregate.c

```diff
@@ -2225,6 +2226,7 @@ agg_object(daos_handle_t ih, vos_iter_entry_t *entry,
 		if (rc < 0) {
 			D_ERROR("oid:"DF_UOID" ds_pool_check_leader failed "
 				DF_RC"\n", DP_UOID(entry->ie_oid), DP_RC(rc));
+			agg_param->ap_obj_skipped = 1;
```

Comment:
This will disable aggregation for degraded mode. I think you need set this if the shard is in rebuilding status, i.e. base this patch on PR-6628.

Reply:
refreshed.